packages/strapi-plugin-users-permissions/admin/src/index.js

```diff
@@ -32,27 +37,75 @@ export default strapi => {
     isRequired: pluginPkg.strapi.required || false,
     layout,
     lifecycles,
-    mainComponent: App,
     name,
     pluginLogo,
     preventComponentRendering: false,
-    reducers,
-    settings: {},
-    trads,
-    menu: {
-      pluginsSectionLinks: [
-        {
-          destination: `/plugins/${pluginId}`,
-          icon,
-          label: {
-            id: `${pluginId}.plugin.name`,
-            defaultMessage: 'Roles & Permissions',
+    reducers: null, // TODO : this is temp
+    settings: {
+      menuSection: {
+        id: pluginId,
+        title: getTrad('plugin.name'),
+        links: [
+          {
+            title: {
+              id: getTrad('HeaderNav.link.roles'),
+              defaultMessage: 'Roles',
+            },
+            name: 'roles',
+            to: `${strapi.settingsBaseURL}/${pluginId}/roles`,
+            Component: () => (
+              <CheckPagePermissions permissions={pluginPermissions.settings}>
```

Comment:
The settings key does not exist in the `permissions.js` file. Maybe you could update this file as follows:

```js
const pluginPermissions = {
  // Roles
  accessRoles: [
    { action: 'plugins::users-permissions.roles.create', subject: null },
    { action: 'plugins::users-permissions.roles.read', subject: null },
    { action: 'plugins::users-permissions.roles.update', subject: null },
  ],
  createRole: [{ action: 'plugins::users-permissions.roles.create', subject: null }],
  deleteRole: [{ action: 'plugins::users-permissions.roles.delete', subject: null }],
  readRoles: [{ action: 'plugins::users-permissions.roles.read', subject: null }],
  updateRole: [{ action: 'plugins::users-permissions.roles.update', subject: null }],

  // AdvancedSettings
  readAdvancedSettings: [
    { action: 'plugins::users-permissions.advanced-settings.read', subject: null },
    { action: 'plugins::users-permissions.advanced-settings.update', subject: null },
  ],
  updateAdvancedSettings: [
    { action: 'plugins::users-permissions.advanced-settings.update', subject: null },
  ],

  // Emails
  readEmailTemplates: [
    { action: 'plugins::users-permissions.email-templates.read', subject: null },
    { action: 'plugins::users-permissions.email-templates.update', subject: null },
  ],
  updateEmailTemplates: [
    { action: 'plugins::users-permissions.email-templates.update', subject: null },
  ],

  // Providers
  readProviders: [
    { action: 'plugins::users-permissions.providers.read', subject: null },
    { action: 'plugins::users-permissions.providers.update', subject: null },
  ],
  updateProviders: [{ action: 'plugins::users-permissions.providers.update', subject: null }],
};

export default pluginPermissions;
```

Reply:
Here you will need to a `Router` for the roles pages so you can't wrap the component using the `CheckPagePermissions` 

`